vllm/model_executor/sampling_metadata.py

```diff
@@ -134,13 +134,62 @@ def __init__(
         num_prompts: int,
         skip_sampler_cpu_output: bool = False,
         reuse_sampling_tensors: bool = False,
+        # Used when multi-step is enabled with chunked-prefill. Refer to
+        # the comment in prepare_multistep_tensors.
+        selected_token_indices_multistep: Optional[torch.Tensor] = None
     ) -> None:
         self.seq_groups = seq_groups
         self.selected_token_indices = selected_token_indices
         self.categorized_sample_indices = categorized_sample_indices
         self.num_prompts = num_prompts
         self.skip_sampler_cpu_output = skip_sampler_cpu_output
         self.reuse_sampling_tensors = reuse_sampling_tensors
+        self.selected_token_indices_multistep = selected_token_indices_multistep
+
+    def prepare_multistep_tensors(self, num_queries: int, device: str,
```

Comment:
I now understand the purpose of this function. Although it simply just creates a `list(range(num_queries))`, you kicked off this function asyncly to hide the latency in `advance_step`. If possible, can we consider the following alternative to simply the code here:

1. Inline `prepare_multistep_tensors` to `advance_step`, but still do async transfer.
2. [Need check] I suppose we don't need a barrier for `self.selected_token_indices`, because the kernel that uses this tensor will need to wait for it.

Reply:
Yeah. I believe we can move this into `multi_step_model_runner`'s advance_step and localize the changes to that file.